### ISSUE TITLE
App "wasn't able to create a security bookmark" (on USB flash drive)

### ISSUE THREAD

**Issue by natevw** [vrunkel/F3XSwift#1]
Thanks for making this handy interface. It worked great a few weeks ago when I tested some SD Cards. Today I got a USB thumbdrive which I thought maybe it would be able to handle also?

But for whatever reason today I am getting an error after I "open" the drive:

> For some reason I wasn't able to create a security bookmark. Thus write access to the selected volume is not possible. Canceling test.

It does write 1,026,048  bytes to one (1.h2w) file.

**Comment by vrunkel** [comment]
Hello Nathan,


Thanks for your feedback. Interesting finding. I don’t have access to a thumb drive. It sounds like the OS revokes access. Which macOS are you running?

Best wishes

Volker

**Comment by natevw** [comment]
> Which macOS are you running?

macOS 10.14.6 (18G9028)

**Comment by dehy** [comment]
Same problem here with macOS 11.4 with a SD Card

**Comment by dehy** [comment]
More information : the error message appears and then we get back to the selection screen but the app is writing the medium in background (more than 1 file, more than 1MB)

**Comment by vrunkel** [comment]
Thx, that might help. Will have soon time to look further into it.

volker

**Comment by errrepe** [comment]
Had the same problem on Tahoe on a SD card. The error pops up but the app still keeps writing 1.07GB files on the background

**Comment by williaty** [comment]
I just encountered this problem today. Tahoe 26.5.1 with today's current build of F3XSwift. Was a workaround to this ever found?

**Comment by vrunkel** [comment]
Hi there,

not to my knowledge. I haven’t run the app myself in the last years, though. Last macOS releases, especially 26 changed a lot of internals around secure file access in sandboxed environments. 

best wishes,
volker

**Comment by ericrhall** [comment]
I too am unable to run this with the same error on 26.5.1. You haven't run it in years? FStoppers just recommended it. Do you plan to revisit this app?

**Comment by vrunkel** [comment]
Nope, havent run it in years. My engineers have built a mechanical sd card testing device for me (and no, never for sale).

I don’t know who/what FStoppers is :| - I have plans to revive the app, but my workload is too high, so in near future it won’t happen. It will need a complete rewrite for all the things changed in the last few macOS and swift versions. part of my high workload since I am maintaining a rather large app for my company and have to do a complete rewrite for this as well. 

Maybe should archive it on github somehow or clearly state the current non-developping state :|

**Comment by williaty** [comment]
For those who end up here asking the same question: The only workaround I've found is to install Homebrew, then install f3 via Homebrew, and run f3 via the command line. That works perfectly fine. No errors. The big downside is that you have to be comfortable using the command line in Terminal.

**Comment by ericrhall** [comment]
Here is the video from fstoppers:
https://youtu.be/h2cFsB7yRWw?si=3FWBrt9B-XrOr60N

I will try F3 via homebrew as suggested by williaty.

Eric

Eric Hall
360-770-5256 cell

**Comment by kult** [comment]
Also getting same issue. Also here from Fstoppers. Vrunkle, this repo is about to get a lot of attention because Fstoppers have a million YouTube subs. I'm sorry to hear that F3XSwift development has been indefinitely paused for years, because it seems to be a great tool for combating the shoddy and counterfeit media that run rampant.
